packages/shared/src/hooks/useQuerySubscription.ts

```diff
@@ -0,0 +1,37 @@
+import type {
+  DefaultError,
+  Mutation,
+  UseMutationOptions,
+} from '@tanstack/react-query';
+import { useMutationSubscription } from './mutationSubscription';
+
+export const useQuerySubscription = <
```

Comment:
Not 100% about the name, it is basically typed wrapper over `useMutationSubscription`, WIP 😆